### PR TITLE
alacritty: use hiPrio (fix #71300)

### LINKED ISSUE #71300: No colour in alacritty when installed via `configuration.nix`
**Describe the bug**
There is no colour in neovim in alacritty if alacritty is installed from `configuration.nix`. `alacritty` has to be installed via `nix-env` for colours to work.

**To Reproduce**
1. Install alacritty in `configuration.nix` only
2. Run neovim: no colours

Please keep in mind that I am using neovim as an example. The problem doesn't seem to be related to it in any way.

**Expected behavior**

Neovim has colours.

### PR DESCRIPTION
**Pull request by Br1ght0ne**
This should allow `alacritty` to override any `terminfo/a/alacritty*` files
provided by `ncurses` with the ones freshly compiled with `tic`.

###### Motivation for this change
Fix https://github.com/NixOS/nixpkgs/issues/71300

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @Mic92 @worldofpeace @NilsIrl

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/74358)

### CODE REVIEW

File: pkgs/applications/misc/alacritty/default.nix [hunk 1]
```diff
@@ -116,5 +116,10 @@ in buildRustPackage rec {
     license = with licenses; [ asl20 ];
     maintainers = with maintainers; [ filalex77 mic92 ];
     platforms = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "aarch64-linux" ];
+
+    /* Give alacritty a higher priority to override any `terminfo/a/alacritty*`
```
Comment (worldofpeace) [inline pkgs/applications/misc/alacritty/default.nix]: Possibly link the issue here that it fixes.

**Comment by NilsIrl** [comment]
I tested it and it doesn't work or fix anything for me.

**Comment by teto** [comment]
Rather than hardcoding the number, I recommend to use the helpers in lib/meta.nix like `hiPrio` that could be used from all-packages.nix: `alacritty = hiPrio (callPackage ...);`

**Comment by filalex77** [comment]
@GrahamcOfBorg build alacritty

**Review by teto — APPROVED** [review]

**Comment by teto** [comment]
@NilsIrl could you retest before I merge please ?

**Comment by NilsIrl** [comment]
Doesn't fix the problem for me

**Comment by Mic92** [comment]
Confirmed, it only merges in the terminfo variants that are not in ncurses.

```
$ ls -la /run/current-system/sw/share/terminfo/a/alacritty*
lrwxrwxrwx 91 root  1 Jan  1970  /run/current-system/sw/share/terminfo/a/alacritty -> /nix/store/yfpiz0n3dykir9r2skbxlvcard9xp275-ncurses-6.1-20190112/share/terminfo/a/alacritty
lrwxrwxrwx 98 root  1 Jan  1970  /run/current-system/sw/share/terminfo/a/alacritty+common -> /nix/store/yfpiz0n3dykir9r2skbxlvcard9xp275-ncurses-6.1-20190112/share/terminfo/a/alacritty+common
lrwxrwxrwx 98 root  1 Jan  1970  /run/current-system/sw/share/terminfo/a/alacritty-direct -> /nix/store/yfpiz0n3dykir9r2skbxlvcard9xp275-ncurses-6.1-20190112/share/terminfo/a/alacritty-direct
```

Right now I switched to TERM=alacritty-direct in `~/.config/alacritty/alacritty.yml`:

```
env:
#  # TERM variable
#  #
#  # This value is used to set the `$TERM` environment variable for
#  # each instance of Alacritty. If it is not present, alacritty will
#  # check the local terminfo database and use 'alacritty' if it is
#  # available, otherwise 'xterm-256color' is used.
#  TERM: xterm-256color
   TERM: alacritty-direct
```

**Comment by filalex77** [comment]
@NilsIrl @Mic92 Thanks a lot for your help.
If that's the case, I'll close the PR for now and update the derivation as soon as the release is out.

**Pull request closed without merging**